Copy_from_upstream: no subprocess call & update_cbom fix for CI.

bhess opened the pull request:
- Refactor update_cbom and update_docs_from_yaml, make them usable via import
- No subprocess call anymore in copy_from_upstream
- CBOM now don't change if there are no algorithm changes
- Workaround for issue in GitPython, caused update_cbom to fail in Github CI. (also noticed in https://github.com/gitpython-developers/GitPython/issues/427)

Pipeline in #1405 should work after this PR.

Fixes #1406 #1407.

<!-- Please give a brief explanation of the purpose of this pull request. -->

<!-- Does this PR resolve any issue?  If so, please reference it using automatic-closing keywords like "Fixes #123." -->

<!-- Please answer the following questions to help manage version and changes across projects. -->

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [ ] Does this PR change the the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in [oqs-provider](https://github.com/open-quantum-safe/oqs-provider), [OQS-OpenSSL](https://github.com/open-quantum-safe/openssl), [OQS-BoringSSL](https://github.com/open-quantum-safe/boringssl), and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

<!-- Once your pull request is ready for review and passing continuous integration tests, please convert from a draft PR to a normal PR, and request a review from one of the OQS core team members. -->

baentsch reviewed (changes requested):
Briefly glanced over the code and it addresses my concerns, Thanks! I have not checked all python lines explicitly, though. Please let me know if you think this is warranted. 

What does need changing, though, is a run of this new code (and incorporation of the resultant changes into the PR) prior to merge: The CI test in #1405 otherwise fails: My expectation is that anyone running `python3 copy_from_upstream.py copy` on this PR (and any future ones) should see no changes when running `git status` afterwards.

bhess commented:
Added a commit with the changes after running copy_from_upstream.. 

`git status` before the update (some algo docs needed to be updated):

```
$ git status
On branch bhe-cbomupd
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   docs/algorithms/kem/classic_mceliece.md
	modified:   docs/algorithms/kem/classic_mceliece.yml
	modified:   docs/algorithms/kem/hqc.md
	modified:   docs/algorithms/kem/hqc.yml
	modified:   docs/algorithms/kem/kyber.md
	modified:   docs/algorithms/kem/kyber.yml
	modified:   docs/algorithms/sig/dilithium.md
	modified:   docs/algorithms/sig/dilithium.yml
```

Running copy_from_upstream again -> git status shows no changes as expected:

```
$ git status
On branch bhe-cbomupd
nothing to commit, working tree clean
```

> Briefly glanced over the code and it addresses my concerns, Thanks! I have not checked all python lines explicitly, though. Please let me know if you think this is warranted.

Thanks, I don't think it's necessary to check all lines. Most of the diffs are just due to putting script code to functions, making them usable via module import. Each Indentation shows up as a diff.
